### PR TITLE
[Narwhal] Keep 10s of inflight certificate broadcasts

### PR DESCRIPTION
**Pull request by mwtian**
## Description 

As the network grows in size with more remote peers, it is useful to tolerate temporary jitters in the latency to deliver certificates.

## Test Plan 

Deploy to private testnet and monitor.

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: narwhal/primary/src/synchronizer.rs [hunk 1]
```diff
@@ -611,82 +616,77 @@ impl Synchronizer {
...
+                Some((cert, resp)) = requests.next() => {
+                    backoff_multiplier = match resp {
+                        Ok(_) => {
+                            0
+                        },
+                        Err(_) => {
+                            if requests.is_empty() {
+                                // Retry broadcasting the latest certificate, to help the network stay alive.
+                                let request = Request::new(SendCertificateRequest { certificate: cert.clone() }).with_timeout(PUSH_TIMEOUT);
+                                requests.push_back(send_certificate(client.clone(),request, cert));                                min(backoff_multiplier * 2 + 1, MAX_BACKOFF_MULTIPLIER)
```
Comment (akichidis) [inline narwhal/primary/src/synchronizer.rs:678]: Let's check indentation on this line - it renders a bit off in my IDE

File: narwhal/primary/src/synchronizer.rs [hunk 2]
```diff
@@ -611,82 +616,77 @@ impl Synchronizer {
...
+                            } else {
+                                // TODO: add backoff and retries for transient & retriable errors.
```
Comment (akichidis) [inline narwhal/primary/src/synchronizer.rs:680]: 👍

**Review by akichidis — APPROVED** [review]
Thanks @mwtian - any specific metrics you expect to see improving here?

File: narwhal/primary/src/synchronizer.rs (hunk 1 above)
Comment (mwtian) [reply narwhal/primary/src/synchronizer.rs]: Weird that `cargo fmt` did not fix this. Updated.

**Pull request merged**